langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -699,6 +699,23 @@ function testCloneWithTypeNumeric7() {
     assert(a2[2], <decimal> 3);
 }
 
+function testCloneWithTypeDecimalToInt() {
+    decimal a = 12.3456;
+    int|error result = a.cloneWithType(int);
+    assert(result is int, true);
+    assert(checkpanic result, 12);
+
+    decimal[] a1 = [1.23, 2.34, 3.45];
```

Comment:
Shall we write a test check for negative cases? Like out of range

Reply:
Also, any idea why code patch coverage is low?

Reply:
The changes can be applied to `byte` & int sub types as well. Added more tests to cover them.